### PR TITLE
Get AEE runner from OpenstackVersion

### PR DESCRIPTION
**Pull request by fao89**
ref: https://github.com/openstack-k8s-operators/ci-framework/pull/1580

### CODE REVIEW

File: tests/roles/backend_services/templates/openstack_version.j2 [hunk 1]
```diff
@@ -5,6 +5,7 @@ metadata:
   namespace: openstack
 spec:
   customContainerImages:
+    ansibleeeImage: {{ ansibleee_runner_img | default('quay.io/openstack-k8s-operators/openstack-ansibleee-runner:latest')}}
```
Comment (cescgina) [inline tests/roles/backend_services/templates/openstack_version.j2:8]:
```suggestion
    {% if not skip_patching_ansibleee_csv is defined or not skip_patching_ansibleee_csv | bool %}
    ansibleeeImage: {{ ansibleee_runner_img | default('quay.io/openstack-k8s-operators/openstack-ansibleee-runner:latest')}}
    {% endif %}
```

**Review by cescgina — CHANGES_REQUESTED** [review]
We need something a bit more complex here. Using the variable `skip_patching_ansibleee_csv` we could skip setting the ansible-runner image and rely on the operator default, which we used when testing beta downstream, we need some jinja templating here to retain the functionality. I left a suggestion inline, but we'd still need to test it

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
the ceph issue is probably related to:
- https://github.com/openstack-k8s-operators/dataplane-operator/pull/896
- https://github.com/openstack-k8s-operators/ci-framework/pull/1757
- https://github.com/openstack-k8s-operators/architecture/pull/247

**Comment by fao89** [comment]
the deployment was completed:
https://logserver.rdoproject.org/69/469/7229889827dd08760087c738a5cbc548955dd5e6/github-check/data-plane-adoption-osp-17-to-extracted-crc-minimal-no-ceph/247a407/controller/ci-framework-data/logs/openstack-k8s-operators-openstack-must-gather/namespaces/openstack/crs/openstackdataplanedeployments.dataplane.openstack.org/openstack.yaml

the failure is unrelated to this PR

**Comment by fao89** [comment]
recheck

**Review by marios — COMMENTED** [review]
looks OK, but wondering if the non-ceph job failure is related? I don't think so but it has not passed on this patch? Perhaps we have a new issue/cix there @cescgina do you know?

i hit this issue when developing the multinode (it was rabbit related at that time, but might be something else in this case)

https://logserver.rdoproject.org/69/469/3bfd3d7b768a9a5cd46985a4bbf5a371bf2195ad/github-check/data-plane-adoption-osp-17-to-extracted-crc-minimal-no-ceph/a056775/controller/data-plane-adoption-tests-repo/data-plane-adoption/tests/logs/test_minimal_out_2024-05-27T21:02:14EDT.log

```
        * TASK [dataplane_adoption : wait for cell1 Nova compute EDPM services version updated] ***
        * fatal: [localhost]: FAILED! => {"attempts": 20, "changed": true, "cmd": "set -euxo pipefail\n\n\nPODIFIED_MARIADB_IP=10.217.5.136\nPODIFIED_CELL1_MARIADB_IP=10.217.5.132\nPODIFIED_DB_ROOT_PASSWORD=\"12345678\"\n\n# The CHARACTER_SET and collation should match the source DB\n# if the do not then it will break foreign key relationships\n# for any tables that are created in the future as part of db sync\nCHARACTER_SET=utf8\nCOLLATION=utf8_general_ci\n\noc exec openstack-cell1-galera-0 -c galera -- mysql -rs -uroot -p$PODIFIED_DB_ROOT_PASSWORD  -e \"select a.version from nova_cell1.services a join nova_cell1.services b where a.version!=b.version and a.binary='nova-compute';\"\n", "delta": "0:00:00.246207", "end": "2024-05-27 21:50:31.918957", "msg": "", "rc": 0, "start": "2024-05-27 21:50:31.672750", "stderr": "+ PODIFIED_MARIADB_IP=10.217.5.136\n+ PODIFIED_CELL1_MARIADB_IP=10.217.5.132\n+ PODIFIED_DB_ROOT_PASSWORD=12345678\n+ CHARACTER_SET=utf8\n+ COLLATION=utf8_general_ci\n+ oc exec openstack-cell1-galera-0 -c galera -- mysql -rs -uroot -p12345678 -e 'select a.version from nova_cell1.services a join nova_cell1.services b where a.version!=b.version and a.binary='\\''nova-compute'\\'';'", "stderr_lines": ["+ PODIFIED_MARIADB_IP=10.217.5.136", "+ PODIFIED_CELL1_MARIADB_IP=10.217.5.132", "+ PODIFIED_DB_ROOT_PASSWORD=12345678", "+ CHARACTER_SET=utf8", "+ COLLATION=utf8_general_ci", "+ oc exec openstack-cell1-galera-0 -c galera -- mysql -rs -uroot -p12345678 -e 'select a.version from nova_cell1.services a join nova_cell1.services b where a.version!=b.version and a.binary='\\''nova-compute'\\'';'"], "stdout": "57", "stdout_lines": ["57"]}

```

comes from there https://github.com/openstack-k8s-operators/data-plane-adoption/blob/2851e6397338ad771d781b263add610d9a32577d/tests/roles/dataplane_adoption/tasks/nova_ffu.yaml#L6

**Comment by marios** [comment]
> looks OK, but wondering if the non-ceph job failure is related? I don't think so but it has not passed on this patch? Perhaps we have a new issue/cix there @cescgina do you know?
> 
> i hit this issue when developing the multinode (it was rabbit related at that time, but might be something else in this case)
> 
> https://logserver.rdoproject.org/69/469/3bfd3d7b768a9a5cd46985a4bbf5a371bf2195ad/github-check/data-plane-adoption-osp-17-to-extracted-crc-minimal-no-ceph/a056775/controller/data-plane-adoption-tests-repo/data-plane-adoption/tests/logs/test_minimal_out_2024-05-27T21:02:14EDT.log
> 
> ```
>         * TASK [dataplane_adoption : wait for cell1 Nova compute EDPM services version updated] ***
>         * fatal: [localhost]: FAILED! => {"attempts": 20, "changed": true, "cmd": "set -euxo pipefail\n\n\nPODIFIED_MARIADB_IP=10.217.5.136\nPODIFIED_CELL1_MARIADB_IP=10.217.5.132\nPODIFIED_DB_ROOT_PASSWORD=\"12345678\"\n\n# The CHARACTER_SET and collation should match the source DB\n# if the do not then it will break foreign key relationships\n# for any tables that are created in the future as part of db sync\nCHARACTER_SET=utf8\nCOLLATION=utf8_general_ci\n\noc exec openstack-cell1-galera-0 -c galera -- mysql -rs -uroot -p$PODIFIED_DB_ROOT_PASSWORD  -e \"select a.version from nova_cell1.services a join nova_cell1.services b where a.version!=b.version and a.binary='nova-compute';\"\n", "delta": "0:00:00.246207", "end": "2024-05-27 21:50:31.918957", "msg": "", "rc": 0, "start": "2024-05-27 21:50:31.672750", "stderr": "+ PODIFIED_MARIADB_IP=10.217.5.136\n+ PODIFIED_CELL1_MARIADB_IP=10.217.5.132\n+ PODIFIED_DB_ROOT_PASSWORD=12345678\n+ CHARACTER_SET=utf8\n+ COLLATION=utf8_general_ci\n+ oc exec openstack-cell1-galera-0 -c galera -- mysql -rs -uroot -p12345678 -e 'select a.version from nova_cell1.services a join nova_cell1.services b where a.version!=b.version and a.binary='\\''nova-compute'\\'';'", "stderr_lines": ["+ PODIFIED_MARIADB_IP=10.217.5.136", "+ PODIFIED_CELL1_MARIADB_IP=10.217.5.132", "+ PODIFIED_DB_ROOT_PASSWORD=12345678", "+ CHARACTER_SET=utf8", "+ COLLATION=utf8_general_ci", "+ oc exec openstack-cell1-galera-0 -c galera -- mysql -rs -uroot -p12345678 -e 'select a.version from nova_cell1.services a join nova_cell1.services b where a.version!=b.version and a.binary='\\''nova-compute'\\'';'"], "stdout": "57", "stdout_lines": ["57"]}
> ```
> 
> comes from there
> 
> https://github.com/openstack-k8s-operators/data-plane-adoption/blob/2851e6397338ad771d781b263add610d9a32577d/tests/roles/dataplane_adoption/tasks/nova_ffu.yaml#L6

same issue found on another pull request so looks like cix 

https://logserver.rdoproject.org/62/462/429efde0f496a6073d322512c0ec554120e4e7fa/github-check/data-plane-adoption-osp-17-to-extracted-crc-minimal-no-ceph/a738c36/controller/data-plane-adoption-tests-repo/data-plane-adoption/tests/logs/test_minimal_out_2024-05-27T13:54:14EDT.log

**Comment by fao89** [comment]
recheck

**Comment by marios** [comment]
recheck 

mainly i want to see if this NODE_FAILURE is a real thing (only 2 from yestreday https://review.rdoproject.org/zuul/builds?job_name=data-plane-adoption-osp-17-to-extracted-crc&skip=0)

I don't think we really need to re-run here - it was tested in testproject and seen to do the right thing. I think we can force merge it today. the non-ceph issue is being tracked with https://issues.redhat.com/browse/OSPCIX-307 which may be fixed now (https://github.com/openstack-k8s-operators/ci-framework/pull/1757)

**Review by cescgina — APPROVED** [review]
Thanks for the changes! I tested this in https://review.rdoproject.org/r/c/testproject/+/53274 and it works. Job is green https://review.rdoproject.org/zuul/build/9c7a3f56623c41248a790136c29a4aea, using ansiblee-runner image from [https://logserver.rdoproject.org/74/53274/2/check/data-plane-adoption-osp-17-to-extracted[…]/controller/ci-framework-data/artifacts/edpm-ansible.yml](https://logserver.rdoproject.org/74/53274/2/check/data-plane-adoption-osp-17-to-extracted-crc/9c7a3f5/controller/ci-framework-data/artifacts/edpm-ansible.yml), as shown in [https://logserver.rdoproject.org/74/53274/2/check/data-plane-adoption-osp-17-to-extracted[…]w6q/configure-network-openstack-openstack-s8w6q-describe](https://logserver.rdoproject.org/74/53274/2/check/data-plane-adoption-osp-17-to-extracted-crc/9c7a3f5/controller/ci-framework-data/logs/openstack-k8s-operators-openstack-must-gather/namespaces/openstack/pods/configure-network-openstack-openstack-s8w6q/configure-network-openstack-openstack-s8w6q-describe) Image:         [quay.rdoproject.org/openstack-k8s-operators/openstack-ansibleee-runner:a6745373e2a7d680c2279345917e6803e352be98](http://quay.rdoproject.org/openstack-k8s-operators/openstack-ansibleee-runner:a6745373e2a7d680c2279345917e6803e352be98)

**Review by jistr — APPROVED** [review]

**Comment by fao89** [comment]
recheck

**Pull request merged**